browser/js/cart/cart.js

```diff
@@ -6,16 +6,27 @@ app.config(function($stateProvider) {
   });
 });
 
-app.controller('CartCtrl', function($scope, CartFactory, ProductFactory) {
+app.controller('CartCtrl', function($scope, $uibModal, CartFactory, ProductFactory) {
 
   $scope.cartInfo = CartFactory.getInfo();
   $scope.isInCart = CartFactory.isInCart;
 
   CartFactory.fetchCart()
-  .then(function(_cart) {
-    $scope.cart = _cart;
-  });
+    .then(function(_cart) {
+      $scope.cart = _cart;
+    });
 
+  $scope.openModal = function(id) {
+    $uibModal.open({
+      templateUrl: 'js/products/product.detail.html',
+      controller: 'ProductDetailCtrl',
```

Comment:
Just note that we MAY want to move this out to a distinct controller in the future...